sycl/test-e2e/Regression/alloc_with_invalid_alignment.cpp

```diff
@@ -0,0 +1,41 @@
+// REQUIRES: level_zero
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <algorithm>
+#include <cassert>
+#include <sycl/queue.hpp>
+#include <sycl/usm.hpp>
+
+// This test verifies that for L0 backend, aligned USM alloc functions return
+// null_ptr when called with alignment values that are not a power-of-2.
```

Comment:
Is this standard behavior? If so, why is it limited to `level_zero`? If not, why do we care?

Reply:
Also, why does this test belong here and why is there no test in the UR repo?

Reply:
Yeah, L0 spec says that alignment values should be powers of 2 and SYCL spec says that null should be returned if that is the case. This was created in response to this issue [here](https://github.com/intel/llvm/issues/12638).

For L0 spec see [here](https://spec.oneapi.io/level-zero/latest/core/api.html#zememallocshared) next to the alignment paramater.
For SYCL spec see [here](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_usm_allocations) in the second paragraph.

Reply:
> Also, why does this test belong here and why is there no test in the UR repo?

I've put here since it is verifying a behaviour mandated by the SYCL spec.  Also, @kbenzie, thoughts on adding a test for this in UR repo? I suppose since this PR does not affect SYCL source code, the test is indirectly testing L0 changes as well but perhaps it could be a good idea to add one there as well.

Reply:
Please re-write the test to verify what SYCL spec mandates and make it backend-agnostic.

Reply:
@aelovikov-intel There is some source code to be tested on the SYCL side such as the logic of checking that the L0 allocation functions return invalid codes and then turning them into nullptr from the point of view of a SYCL user and that's what this test is doing. Granted, it's minuscule logic, but I do think it should be tested.

Reply:
What I'm saying is that `REQUIRES: level_zero` is totally unnecessary. Also, if some backend can align allocation to 17 bytes then it would be valid to return that instead of `nullptr` (although I can't imagine why would that happen). We should also expand this test for big power-of-two requests as well and verify that either required alignment is satisfied or `nullptr` is returned.

Reply:
> What I'm saying is that `REQUIRES: level_zero` is totally unnecessary. Also, if some backend can align allocation to 17 bytes then it would be valid to return that instead of `nullptr` (although I can't imagine why would that happen). We should also expand this test for big power-of-two requests as well and verify that either required alignment is satisfied or `nullptr` is returned.

Ok, I will make these changes.